### ISSUE TITLE
Iesi.Collections package dependency range causes warning

### ISSUE THREAD

**Issue by bording** [nhibernate/nhibernate-core#1430]
There is a mismatch in the dependency range specified for the Iesi.Collections package and the actual versions available for that package.

![image](https://user-images.githubusercontent.com/753669/32452260-0bba7176-c2e7-11e7-9c14-6af63f329302.png)

The minimum version is 4.0.0, however there is no 4.0.0 version of the package:

![image](https://user-images.githubusercontent.com/753669/32452314-2b952694-c2e7-11e7-98ea-1d3132d08f3d.png)

When using the NHibernate package with a project that uses the new NuGet transitive restore capabilites provided by using package references instead of packages.config, the following warning occurs during restore:

```
Warning	NU1603	NHibernate 4.0.0.4000 depends on Iesi.Collections (>= 4.0.0 && < 5.0.0) but Iesi.Collections 4.0.0 was not found. An approximate best match of Iesi.Collections 4.0.0.4000 was resolved.
```

This warning can easily be avoided by bumping the minimum version range to 4.0.0.4000, or even better, 4.0.2, since that is the first release to drop the extra version component.

**Comment by fredericDelaporte** [comment]
I think it should be bumped to 4.0.2. I had troubles upgrading a test project which has retained the previous Iesi.Collection and then was failing a bit needlessly due to assembly version mismatch on this dependency. (The binding redirect was also missing.)

**Comment by hazzik** [comment]
Shall be already fixed by #1404
